src/tool/python/code_writers.h

```diff
@@ -635,44 +569,43 @@ static PyType_Spec @_Type_spec =
     }
 
 )");
-            bind_method_overloads<write_class_constructor_overload>(w, constructors);
-        }
+            w.write("    Py_ssize_t arg_count = PyTuple_Size(args);\n\n");
 
-        w.write("}\n");
-    }
+            bool first{ true };
+            for (auto&& m : constructors)
+            {
+                method_signature signature{ m };
 
-    void write_class_property(writer& w, Property const& prop)
-    {
-        XLANG_ASSERT(!is_ptype(prop.Parent()));
+                w.write("    ");
 
-        auto property_methods = get_property_methods(prop);
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    w.write("else ");
+                }
 
-        {
-            auto format = R"(
-static PyObject* @_%(%* self, void* /*unused*/)
-{
-%}
+                auto format = R"(if (arg_count == %)
+    {
 )";
-            w.write(format, 
-                prop.Parent().TypeName(),
-                property_methods.get.Name(),
-                bind<write_winrt_wrapper>(prop.Parent()),
-                bind<write_property_get_body>("self->obj", prop.Name()));
-        }
+                w.write(format, count_in_param(signature.params()));
+                write_class_constructor_overload(w, m, signature);
+                w.write("    }\n");
+            }
 
-        if (property_methods.set)
-        {
-            auto format = R"(
-static int @_%(%* self, PyObject* value, void* /*unused*/)
-{
-%}
-)";
-            w.write(format,
-                prop.Parent().TypeName(),
-                property_methods.set.Name(),
-                bind<write_winrt_wrapper>(prop.Parent()),
-                bind<write_property_set_body>("self->obj", prop.Name(), prop.Type().Type()));
+            w.write(R"(    else if (arg_count == -1)
+    {
+        return nullptr; 
+    }
+
+    PyErr_SetString(PyExc_RuntimeError, "Invalid parameter count");

```

Comment:
Would suggest matching the usual error type and message (though if/when you switch to `PyArg_ParseTupleAndKeywords` for parsing, you'll get this for free).

Reply:
It looks like `PyArg_ParseTupleAndKeywords` uses `PyErr_BadInternalCall` which seems like a mistake. Will switch over to setting `PyExc_TypeError` instead of `PyExc_RuntimeError` to mirror the behavior of `PyErr_BadArgument` (but calling directly so I can pass my own message instead of the hardcoded one passed by `PyErr_BadArgument`)

Reply:
It should only do that if _you_ provide the wrong arguments. It shouldn't do that when the caller provides the wrong ones (otherwise all of Python would be raising the wrong error).